File: Nth node from end.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
def nth_from_end(head, n):
    first = head
    for _ in range(n):
        if not first:
            return -1
        first = first.next

    second = head
    while first:
        first = first.next
        second = second.next
    return second.data

File: Nth node from end.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

struct Node {
    int data;
    struct Node* next;
};

struct Node* newNode(int data) {
    struct Node* node = (struct Node*)malloc(sizeof(struct Node));
    node->data = data;
    node->next = NULL;
    return node;
}
int nthFromEnd(struct Node* head, int n) {
    struct Node* first = head, *second = head;
    for (int i = 0; i < n; i++) {
        if (!first) return -1;
        first = first->next;
    }
    while (first) {
        first = first->next;
        second = second->next;
    }
    return second->data;
}
